include/openenclave/internal/sgx/extradata.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_INTERNAL_SGX_EXTRADATA_H
+#define _OE_INTERNAL_SGX_EXTRADATA_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * May be overriden by the host application to add additional enclave data pages
+ * immediately before the enclave heap. The hook will be invoked by the loader
+ * twice: In the first time, the loader constructs a dummy **arg** and passes
+ * **baseaddr** as zero, expecting the hook to invoke oe_load_extra_enclave_data
+ * and returns the total size of extra data (will be stored as part of **arg**).
+ * In the second time, the loader contrusts the **arg** with necessary
+ * parameters and passes the **baseaddr** as the starting address to which the
+ * extra data will be loaded, expecting the hook to invoke
+ * oe_load_extra_enclave_data to load each extra data pages.
+ */
+oe_result_t oe_load_extra_enclave_data_hook(void* arg, uint64_t baseaddr);
```

Comment:
OE SDK hard-codes the type of `arg` to `oe_load_extra_enclave_data_hook_arg_t`. There is no way for a host application to customize it. Therefore, `oe_load_extra_enclave_data_hook_arg_t` needs to be part of the `extradata` interface. `arg` needs to be strongly typed.

Reply:
done